### PR TITLE
Add unit tests for `String::parse_url()`

### PR DESCRIPTION
**Pull request by pafuent**
This PR aims to help "fix" https://github.com/godotengine/godot/issues/43440

PLEASE pay special attention to port invalid URLs test. Current behavior returns the port value even when `String::parse_url()` is returning `Error::ERR_INVALID_PARAMETER`
IMHO `String::parse_url()` shouldn't update port parameter if is invalid. If everyone agree I can change the code to do so.

### CODE REVIEW

**Comment by pafuent** [comment]
I'm not sure if the process of submitting a PR involves saying that it's ready for review, but just in case, I'm saying it.

**Review by Geometror — COMMENTED** [review]
Looks fine to me!
What remains is squashing your commits into one.
Regarding the behavior when the port range is invalid: I'm fine with either way, but I think for now we should leave it.  The user should check the returned Error anyways, not the port value.

**Review by Geometror — APPROVED** [review]

File: tests/core/string/test_string.h [hunk 1]
```diff
@@ -1988,6 +1988,48 @@ TEST_CASE("[String] Variant ptr indexed set") {
...
+	do {                                                                                                                  \
+		int port;                                                                                                         \
+		String url(m_url_to_parse), schema, host, path;                                                                   \
+                                                                                                                          \
+		CHECK_EQ(m_expected_error, url.parse_url(schema, host, port, path));                                              \
+		CHECK_EQ(m_expected_schema, schema);                                                                              \
+		CHECK_EQ(m_expected_host, host);                                                                                  \
+		CHECK_EQ(m_expected_path, path);                                                                                  \
+		CHECK_EQ(m_expected_port, port);                                                                                  \
+	} while (false)
```
Comment (Calinou) [inline tests/core/string/test_string.h:2002]: We tend to use this approach for function-like macros instead:

https://github.com/godotengine/godot/blob/e3213aaef5e0e72b8272e65d989d3d8222be17ca/core/error/error_macros.h#L669-L677

See https://github.com/godotengine/godot/pull/60613 as for why this is used over `do { ... } while (false)`.

**Comment by akien-mga** [comment]
Needs a rebase after #92237.

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**